lib/pbench/test/unit/server/query_apis/commons.py

```diff
@@ -50,26 +53,48 @@ def _setup(
         self.bad_date_payload = bad_date_payload
         self.error_payload = error_payload
         self.empty_es_response_payload = empty_es_response_payload
+        self.index_prefix = index_prefix
+        self.index_version = index_version
 
-    def build_index(self, server_config, dates):
+    def build_index(self, server_config=None, find_metadata=None, dates=[]):
```

Comment:
Yeah I did this to build the index in one place, instead of having 2 separate functions. Otherwise we would have to modify other test files by adding an `if-else` statements to check if the index need to built using the metadata or the client provided dates. e.g some of our common test cases like `test_http_error` or `test_http_exception` need to determine if they should get the index from metadata or from dates as our previous API still depends on start and end dates.